app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt

```diff
@@ -4,16 +4,24 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import com.github.se.polyfit.data.remote.firebase.MealFirebaseRepository
+import com.github.se.polyfit.model.data.User
 import com.github.se.polyfit.model.ingredient.Ingredient
 import com.github.se.polyfit.model.meal.Meal
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
 
-class MealViewModel(
-    private val userId: String,
-    firebaseID: String = "",
-    var initialMeal: Meal? = null,
-    private val mealRepo: MealFirebaseRepository = MealFirebaseRepository(userId)
-) : ViewModel() {
+@HiltViewModel
+class MealViewModel
+@Inject
+constructor(private val user: User, private val mealRepo: MealFirebaseRepository) : ViewModel() {
   // after friday use hilt dependency injection to make code cleaner, for now i guess this is ok
+  private var initialMeal: Meal? = null
+  private var firebaseID: String = ""
+
```

Comment:
The way everything is set up now, `MealViewModel` will be reused, and so the `init` will only be called once. Is there ever going to be a case when we have the `firebaseId` and not the meal? Because the `_meal` directly contains the var  `isComplete`, `firebaseID`, and `meal`. The more I look at it, the more I think that we can simplify this class by only keeping track of the variable `_meal`. Maybe only the var `isComplete` is useful ui.

Reply:
When we get to having an overview screen that shows all meals of the day, we will use new view models each time there. But, you're right that we would have the meal already. I think we can simplify the init heavily, remove firebaseID, and allow a ViewModel to be created with an initial meal still

Reply:
yeah iscomplete is helpful for the UI